dashboard/engines/marketing/config/routes.rb

```diff
@@ -0,0 +1,5 @@
+Marketing::Engine.routes.draw do
+  namespace :teacher do
+    resources :promotions, only: [:show], to: 'promotions#show'
```

Comment:
> after moving it into the Teacher namespace, I had to add to: 'promotions#show' in the routes.rb to get it to find the correct controller, even though it seems like this should be automatic.

@carl-codeorg It should be automatic, and I believe it is. Could you please share more details about the issue?

```suggestion
    resources :promotions, only: [:show]
```
![Screenshot 2025-03-13 at 14 53 09](https://github.com/user-attachments/assets/def43c5f-5a4e-4a57-b560-79103a671c35)

Reply:
Hmm after testing this again it is working... I might be going crazy. Thanks!